core/lib/Cardano/Address/Script.hs

```diff
@@ -90,6 +95,8 @@ data Script
     | RequireAllOf ![Script]
     | RequireAnyOf ![Script]
     | RequireSomeOf Word8 ![Script]
+    | ValidFromSlot Natural
+    | ValidUntilSlot Natural
     deriving stock (Generic, Show, Eq)
```

Comment:
Might be a good opportunity to add our lovely `Ord` instance to `Script` and `ScriptHash` and `KeyHash`don't you think :sweat_smile: ?

Reply:
introduced `foldScript`, added `Ord` to `ScriptHash` and `KeyHash` to avoid things that were needed when adding script pool to  SeqState